galpy/potential/TwoPowerSphericalPotential.py

```diff
@@ -155,15 +185,119 @@ def _zforce(self,R,z,phi=0.,t=0.,_forceFloatEval=False):
+    #              a**(-3 + beta) * (R**2 + z**2)**(1 - beta / 2.) *
+    #              ((z**2 * (a * (alpha - beta) - r * (beta - 2)) +
+    #                R**2 * (r * (beta - 2) *
+    #                        (beta - 1) + 2 * a * (beta - alpha))
+    #                ) * hyp2f1(-2 + beta, beta - alpha + 1, beta,
+    #                           -(a / r)) / Gamma(beta) +
+    #               (a * (R**2 * (2 + alpha * (-3 + beta)) - z**2 * (beta - 2)) +
+    #                r * (-z**2 + R**2 * (beta - 1)) * (beta - 2)) *
+    #               (-2 + beta) * hyp2f1(beta - 1, beta - alpha + 1, beta,
+    #                                    -(a / r)) / Gamma(beta))))
+
+    def _z2deriv(self,R,z,phi=0.,t=0.,_forceFloatEval=False):
+        """
+        NAME:
+           _z2deriv
+        PURPOSE:
+           evaluate the second vertical derivative for this potential
+        INPUT:
+           R - Galactocentric cylindrical radius
+           z - vertical height
+           phi - azimuth
+           t- time
+        OUTPUT:
+           the second vertical derivative
+        HISTORY:
+           2012-07-26 - Written - Bovy (IAS@MPIA)
+           2019-10-20 - Edited - Starkman (UofT)
+        """
+        return self._R2deriv(numpy.fabs(z),R,phi=phi,t=t,
+                             _forceFloatEval=_forceFloatEval) #Spherical potential
+
+
+    # def _Rzderiv(self,R,z,phi=0.,t=0.,_forceFloatEval=False):
+    #     """
+    #     NAME:
+    #        _Rzderiv
+    #     PURPOSE:
+    #        evaluate the mixed R,z derivative for this potential
+    #     INPUT:
+    #        R - Galactocentric cylindrical radius
+    #        z - vertical height
+    #        phi - azimuth
+    #        t - time
+    #     OUTPUT:
+    #        d2phi/dR/dz
+    #     HISTORY:
+    #        2019-10-20 - Written - Starkman (UofT)
+    #     """
+    #     if not _forceFloatEval and self.integerSelf is not None:
+    #         return self.integerSelf._Rzderiv(R,z,phi=phi,t=t)
+    #     # else:
+    #     a, alpha, beta = self.a, self.alpha, self.beta
+    #     r= sqrt(R**2.+z**2.)
+
+    #     if beta == 3.:
+    #         # Need to have derivatives of hyp2f1
+    #         raise ValueError('Not yet implemented')
+
+    #     if beta == 5.:
+    #         # Need to have derivatives of hyp2f1
+    #         raise ValueError('Not yet implemented')
+
+    #     return (R * z * Gamma(beta - 3) / r**5 *
+    #             ((-3 * Gamma(3 - alpha)) / Gamma(beta - alpha) +
+    #              a**(beta - 3) * r**(2 - beta) *
+    #              ((r * (beta - 2) * beta + 3 * a * (-alpha + beta)) *
+    #               hyp2f1(beta - 2, beta - alpha + 1, beta,
+    #                      -(a / r)) / Gamma(beta) +
+    #               (-2 + beta) * (r * (-2 + beta) * beta +
+    #                              a * (alpha * (-3 + beta) + beta)) *
+    #               hyp2f1(beta - 1, beta - alpha + 1, beta,
+    #                      -(a / r)) / Gamma(beta))))
+
```

Comment:
Not sure that we should keep this if it's going to be commented out

Reply:
I've solved the special cases in Mathematica, but they involve partial derivatives of the hypergeometric function, which is not in scipy special functions.

Reply:
This is gone?

Reply:
deleted. Will (not) show up in next push.